src/generic_core/user.ts

```diff
@@ -57,23 +51,17 @@ module Core {
     private clientToInstanceMap_ :{ [clientId :string] :string };
     private instanceToClientMap_ :{ [instanceId :string] :string };
 
-    public static create = (network :Social.Network,
-                            userId  :string) : Promise<User> => {
-      var user = new User(network, userId);
-      return new Promise((fulfill, reject) => {
-        storage.load<UserState>(user.getStorePath()).then((state) => {
-          user.restoreState(state).then(fulfill.bind({}, user),
-                                        fulfill.bind({}, user));
-        }, (e) => {
-          // User not found in storage - we should fulfill the create promise
-          // anyway as this is not an error.
-          fulfill(user);
-        }).catch((e) => {
-          console.error('Uncaught error in User.create: ' + e);
-          reject(user);
-        });
-      });
-    }
+    private fulfillStorageLoad_ : () => void;
+    public onceLoaded_ : Promise<void> = new Promise<void>((F, R) => {
```

Comment:
For this one I think it should just really be public with no _, to match the other onceLoaded in remote-instance.ts

Also for things that are needed by tests but otherwise should be private, I think using both private and _ is good, and we can access them using something like myObj['privateProperty_']